### DISCUSSION TITLE
[SOLVED] Two JavaScript newbie questions for article filtering (whole words / case-sensitivity)

### DISCUSSION THREAD

**Discussion by PMTabs** [martinrotter/rssguard#1264] [Q&A] [answered]
Hello friends, I hope you can help me out with these questions.

1. How do I tell a whitelist/blacklist to match **whole words only**? For example, if I filter "ABCD", it will cause "ABDCEFG" to also be captured by the filter. I've looked into regex word boundaries, using /b, but it doesn't seem to work (or rather, I don't know the correct way to implement it). I should also caveat this by saying I don't want the filter to match only articles with one specific word and no others. Just that the matched word should be _whole_.
2. How do I make the whitelist/blacklist words be **case-insensitive**? For example, a filter of "abcd" should also capture "ABCD" and "AbCd". Currently I have to make at least 3 filter rules for each entry: one with initial capitalisation, one in all caps, and one entirely in lowercase 🤣 That's how much I suck at this. When it comes to phrases with multiple words, I start losing my mind, which is why I came here for help!

Thanks for your time.

**Comment by martinrotter** [comment, marked as answer]
Hi.
1. Yes, you have to use "\b" word boundary syntax. Check below example for better understanding.
2. Yes, it is possible with "i" modifier, see below example, example will only match "XXYY" or "xxyy" as separate words.

```js
const regex = /\bXXYY\b/i;

function filterMessage() {
  if (regex.test(msg.title)) {
    // matched!!!
  }

  return MessageObject.Accept;
}
```

**Reply by PMTabs** [reply]
Thank you so much!

Sorry for the late reply, for some reason I wasn't notified of your response.

Cheers!

EDIT: Sorry for my ignorance, but is there a way to combine multiple words into a single expression? For example:

```
const regex = (/\bWord1\b/i|/\bWord2\b/i);

function filterMessage() {
  if (regex.test(msg.title)) {
    // matched!!!
  }

  return MessageObject.Accept;
}
```

The code above doesn't seem to work for me, so I assume there's some weirdness with the syntax I'm missing.

**Reply by martinrotter** [reply]
```js
const regex = /\b(Word1|Word2|Word3)\b/i;

function filterMessage() {
  if (regex.test(msg.title)) {
    // matched!!!
  }

  return MessageObject.Accept;
}
```
